src/index.ts

```diff
@@ -21,38 +24,54 @@ function makeBase58CheckEncoder(p2pkhVersion: number, p2shVersion: number): (dat
         ) {
           throw Error('Unrecognised address format');
         }
-        addr = Buffer.concat([Buffer.from([p2pkhVersion]), data.slice(3, 3 + data.readUInt8(2))]);
+        if (p2pkhVersion[0] instanceof Array) {
+          addr = Buffer.concat([Buffer.from(p2pkhVersion[0]), data.slice(3, 3 + data.readUInt8(2))]);
+          return bs58Encode(addr);
+        }
+        addr = Buffer.concat([Buffer.from([p2pkhVersion[0]]), data.slice(3, 3 + data.readUInt8(2))]);
         // @ts-ignore
         return bs58Encode(addr);
       case 0xa9: // P2SH: OP_HASH160 <scriptHash> OP_EQUAL
         if (data.readUInt8(data.length - 1) !== 0x87) {
           throw Error('Unrecognised address format');
         }
-        addr = Buffer.concat([Buffer.from([p2shVersion]), data.slice(2, 2 + data.readUInt8(1))]);
+        if (p2shVersion[0] instanceof Array) {
+          addr = Buffer.concat([Buffer.from(p2shVersion[0].slice(0, 2)), data.slice(2, 2 + data.readUInt8(1))]);
```

Comment:
Why slice the array, rather than require it be a certain length?

Reply:
Again, this part is not done by me. slice seems okay to me. What else do you suggest when you say (`require it be a certain length`). Just to check `p2shVersion[0] === 2`?